crypto/fipsmodule/service_indicator/service_indicator_test.cc

```diff
@@ -68,56 +68,89 @@ TEST(ServiceIndicatorTest, BasicTest) {
+
+  // Call an approved service wrapped in an if statement.
+  return_val = 0;
+  CALL_SERVICE_AND_CHECK_APPROVED(approved,
+    if(EVP_AEAD_CTX_seal(aead_ctx.get(), output, &out_len, sizeof(output),
+         nullptr, 0, kPlaintext, sizeof(kPlaintext), nullptr, 0) == 1)
+    {
+      return_val = 1;
+    }
+  );
+  ASSERT_EQ(return_val, 1);
+  ASSERT_EQ(approved, AWSLC_APPROVED);
+
+  // Fail an approved service on purpose.
+  return_val = 0;
+  CALL_SERVICE_AND_CHECK_APPROVED(approved, return_val = EVP_AEAD_CTX_seal(aead_ctx.get(),
+          output, &out_len, 0, nullptr, 0, kPlaintext, sizeof(kPlaintext), nullptr, 0));
+  ASSERT_EQ(return_val, 0);
+  ASSERT_EQ(approved, AWSLC_NOT_APPROVED);
+
+  // Fail an approved service on purpose while wrapped in an if statement.
+  return_val = 0;
+  CALL_SERVICE_AND_CHECK_APPROVED(approved,
+    if(EVP_AEAD_CTX_seal(aead_ctx.get(), output, &out_len, 0,
+        nullptr, 0, kPlaintext, sizeof(kPlaintext), nullptr, 0) == 1)
+    {
+      return_val = 1;
+    }
+  );
+  ASSERT_EQ(return_val, 0);
+  ASSERT_EQ(approved, AWSLC_NOT_APPROVED);
+
+  // Call a non-approved service.
   ASSERT_TRUE(EVP_AEAD_CTX_init(aead_ctx.get(), EVP_aead_aes_128_gcm(),
                                 kAESKey, sizeof(kAESKey), 0, nullptr));
   CALL_SERVICE_AND_CHECK_APPROVED(approved, EVP_AEAD_CTX_seal(aead_ctx.get(),
           output, &out_len, sizeof(output), nonce, EVP_AEAD_nonce_length(EVP_aead_aes_128_gcm()),
           kPlaintext, sizeof(kPlaintext), nullptr, 0));
-  ASSERT_FALSE(approved);
+  ASSERT_EQ(approved, AWSLC_NOT_APPROVED);
 }
 
 TEST(ServiceIndicatorTest, AESCBC) {
   int approved = AWSLC_NOT_APPROVED;
   AES_KEY aes_key;
   uint8_t aes_iv[16];
   uint8_t output[256];
-
   // AES-CBC Encryption KAT
   memcpy(aes_iv, kAESIV, sizeof(kAESIV));
   ASSERT_TRUE(AES_set_encrypt_key(kAESKey, 8 * sizeof(kAESKey), &aes_key) == 0);
-  CALL_SERVICE_AND_CHECK_APPROVED(approved,AES_cbc_encrypt(kPlaintext, output,
+  CALL_SERVICE_AND_CHECK_APPROVED(approved, AES_cbc_encrypt(kPlaintext, output,
                               sizeof(kPlaintext), &aes_key, aes_iv, AES_ENCRYPT));
   ASSERT_TRUE(check_test(kAESCBCCiphertext, output, sizeof(kAESCBCCiphertext),
                   "AES-CBC Encryption KAT"));
-  ASSERT_TRUE(approved);
+  ASSERT_EQ(approved, AWSLC_APPROVED);
 
   // AES-CBC Decryption KAT
   memcpy(aes_iv, kAESIV, sizeof(kAESIV));
   ASSERT_TRUE(AES_set_decrypt_key(kAESKey, 8 * sizeof(kAESKey), &aes_key) == 0);
-  CALL_SERVICE_AND_CHECK_APPROVED(approved,AES_cbc_encrypt(kAESCBCCiphertext, output,
+  CALL_SERVICE_AND_CHECK_APPROVED(approved, AES_cbc_encrypt(kAESCBCCiphertext, output,
                         sizeof(kAESCBCCiphertext), &aes_key, aes_iv, AES_DECRYPT));
   ASSERT_TRUE(check_test(kPlaintext, output, sizeof(kPlaintext),
                   "AES-CBC Decryption KAT"));
-  ASSERT_TRUE(approved);
+  ASSERT_EQ(approved, AWSLC_APPROVED);
 }
 
 TEST(ServiceIndicatorTest, AESGCM) {
```

Comment:
I think it would be good to have a test here for the non-approved use of AES-GCM with an external IV here. May be also the one in BasicTest can be changed to something obviously non-approved, for example, AEAD with ChaCha20Poly1305 (is that possible?), or any other non-approved service and explain in the comment that/why it's not approved. This second ask can wait till later, but then on l.123, clarify why it's not approved.